src/Resource/MoipResource.php

```diff
@@ -134,24 +134,71 @@ public function jsonSerialize()
     }
 
     /**
-     * Find by path.
+     * Execute a http request. If payload == null no body will be sent. Empty body ('{}') is supported by sending a
+     * empty stdClass.
      *
-     * @param string $path
+     * @param            $path
+     * @param            $method
+     * @param mixed|null $payload
+     *
+     * @throws Exceptions\ValidationException  if the API returns a 4xx http status code. Usually means invalid data was sent.
+     * @throws Exceptions\UnautorizedException if the API returns a 401 http status code. Check API token and key
+     * @throws Exceptions\UnexpectedException  if the API returns a 500 http status code, this is not suppose to happen. Please report the error to moip
      *
      * @return stdClass
      */
-    public function getByPath($path)
+    protected function httpRequest($path, $method, $payload = null)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method httpRequest() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10.](https://www.codacy.com/app/jeancesargarcia/moip-sdk-php/file/2076273159/issues/source?bid=2821642&fileBranchId=3138004#l150)